saleor/order/tasks.py

```diff
@@ -21,3 +32,58 @@ def send_order_updated(order_ids):
     manager = get_plugins_manager()
     for order in Order.objects.filter(id__in=order_ids):
         manager.order_updated(order)
+
+
+def _batch_ids(iterable, batch_size=1):
+    length = len(iterable)
+    for index in range(0, length, batch_size):
+        yield iterable[index : min(index + batch_size, length)]
+
+
+def _queryset_in_batches(queryset):
+    ids = queryset.values_list("id", flat=True)
+    for ids_batch in _batch_ids(ids, 1000):
+        yield ids_batch
+
+
+def _expire_orders(manager, now):
+    qs = Order.objects.filter(
+        status=OrderStatus.UNCONFIRMED,
+        channel__expire_orders_after__isnull=False,
+        total_charged_amount=Decimal(0),
+    ).select_related("channel")
```

Comment:
No need to have select here. You use the `qs` only to fetch order's IDs.

Reply:
Also I am not conviced that the query is safe enough. 
We should take into account only the orders that are unconfirmed, without any transaction and without any active payment attached to it. This should give us an orders created by "order-create-from-checkout" mutations.
You should also double check, what kind of the status received the draft order when it is marked as a completed (with and without active auto-confirm ).

Reply:
Also, we should test this with the database that doesn't have any order with transaction to be sure, that we don't break anything related to the old payments or order created from draft, or order created from replacment

Reply:
`quetset_in_batches` from migration has `Input queryset should be sorted be pk.` as a requirement. 
Should we also add the same requirement to yours `_queryset_in_batches`?